### PR TITLE
fix(dashboard): point rule pass-rate widget at existing /?rule= filter

### PR DESCRIPTION
**Pull request by sydlexius**
## Summary

- Slice 2 (#1719) shipped the per-rule pass-rate widget with a placeholder href to `/rules/{id}/results` (404). On audit, the dashboard action queue flyout already filters by rule (URL key `rule`, read by `handleIndex`) -- so the carry-over "build a new page" task is a non-gap.
- Two-line href change in `health_summary.templ` (+ regenerated templ.go) + flipped one assertion in `handlers_report_test.go::TestHandleReportHealth_HTMX_WithRuleResults`.
- Drill-down now lands on the lifecycle-state action queue (rule_violations) -- which is the right "act on failures" surface anyway.

## Linked issue

Closes #1720
Part of #699

## Pre-flight checklist

- [x] Pre-push gate green locally (hook ran on push).
- [x] Code review pass: skipped local CR for a 2-line href change (mechanical).
- [x] Commits squashed: single commit.
- [x] At least one label set: enhancement, ui, docs: not-required.
- [x] Docs label decision made: `docs: not-required` (no user-visible doc surface; widget destination change is self-evident).
- [ ] Screenshot attached: link target changes only; no visual change.
- [x] UAT performed: not run -- href change verified via the existing TestHandleReportHealth_HTMX_WithRuleResults assertion (updated to match new URL).
- [x] OpenAPI spec: no API surface change.
- [x] `templ generate` re-run and `*_templ.go` committed.

## Test plan

- [x] `go test -race ./internal/api/` passes locally.
- [x] Pre-push gate passes locally (auto-run by hook on push).
- [ ] Manual UAT steps:
  - [ ] Open dashboard, click a rule in the per-rule pass-rate widget; confirm it lands on dashboard with that rule pre-selected in the action queue filter (URL ends `/?rule=<id>`).
- [ ] Reviewer follow-ups: none

## Summary by CodeRabbit

* **Bug Fixes**
  * Updated health summary rule links to navigate using the dashboard's rule filtering interface instead of a dedicated results page.

[Review Change Stack](https://app.coderabbit.ai/change-stack/sydlexius/stillwater/pull/1721?utm_source=github_walkthrough&utm_medium=github&utm_campaign=change_stack)

<details open><summary><h3>Greptile Summary</h3></summary>

Fixes the per-rule pass-rate widget's broken drill-down link by redirecting it from the non-existent `/rules/{id}/results` route to the existing dashboard filter at `/?rule=<id>`, and adds `url.QueryEscape` to protect against reserved characters in rule IDs.

- `health_summary.templ` and its regenerated `health_summary_templ.go` swap the `href` format and import `net/url` for proper query-string encoding.
- `handlers_report_test.go` updates the existing URL assertion and adds a new test (`TestHandleReportHealth_HTMX_DrillDownEscapesRuleID`) that seeds a rule ID containing `:`, `/`, and `&` and verifies the encoded form is emitted while the raw form is absent.
</details>

<details open><summary><h3>Confidence Score: 5/5</h3></summary>

Safe to merge; the change is a two-line URL fix with correct query-string encoding and solid test coverage including a reserved-character edge case.

The href destination is now an existing, tested route. url.QueryEscape correctly encodes reserved characters in rule IDs before they reach the URL. The regenerated templ.go matches the source, and the new test covers both the happy path and the raw-form regression guard.

No files require special attention.
</details>

<details open><summary><h3>Important Files Changed</h3></summary>

| Filename | Overview |
|----------|----------|
| web/templates/health_summary.templ | Href changed from broken /rules/{id}/results to /?rule={encoded_id}; url.QueryEscape added; change is correct and safe. |
| web/templates/health_summary_templ.go | Regenerated templ output; URL construction and line-number metadata updated to match the .templ source; no manual edits. |
| internal/api/handlers_report_test.go | Existing assertion updated to the new URL format; new DrillDownEscapesRuleID test correctly validates both positive (encoded form present) and negative (raw form absent) conditions. |

</details>

</details>

<details open><summary><h3>Sequence Diagram</h3></summary>

```mermaid
sequenceDiagram
    participant User
    participant Dashboard
    participant HealthWidget as Health Summary Widget
    participant ActionQueue as Action Queue

    User->>Dashboard: Open dashboard
    Dashboard->>HealthWidget: GET /api/v1/reports/health (HX-Request)
    HealthWidget-->>Dashboard: "Rule pass-rate links with href BasePath/?rule=encodedID"
    User->>HealthWidget: Click rule row
    HealthWidget->>ActionQueue: "Navigate to /?rule=encodedID"
    ActionQueue-->>User: Dashboard filtered by rule via handleIndex
```
</details>

<sub>Reviews (2): Last reviewed commit: ["fix: URL-escape rule ID in drill-down hr..."](https://github.com/sydlexius/stillwater/commit/e0e4aedc321314f406673e2a965a3bf99b3a5c65) | [Re-trigger Greptile](https://app.greptile.com/api/retrigger?id=34310255)</sub>

### CODE REVIEW

File: web/templates/health_summary.templ [hunk 1]
```diff
@@ -93,7 +93,7 @@ templ HealthSummaryFragment(data HealthSummaryData) {
 			<div class="space-y-2">
 				for _, pr := range data.RulePassRates {
 					<a
-						href={ templ.SafeURL(fmt.Sprintf("%s/rules/%s/results", data.BasePath, pr.RuleID)) }
+						href={ templ.SafeURL(fmt.Sprintf("%s/?rule=%s", data.BasePath, pr.RuleID)) }
```
[the remark this answers is not in the document]
Comment (sydlexius) [reply web/templates/health_summary.templ]: Fixed in e0e4aed. Wrapped `pr.RuleID` with `url.QueryEscape` in the drill-down href.

File: internal/api/handlers_report_test.go [hunk 2]
```diff
@@ -224,8 +224,8 @@ func TestHandleReportHealth_HTMX_WithRuleResults(t *testing.T) {
 		t.Fatalf("status = %d body=%s", w.Code, w.Body.String())
 	}
 	body := w.Body.String()
-	if !strings.Contains(body, "/musicbrainz/rules/rr-htmx/results") {
-		t.Errorf("expected BasePath-prefixed drill-down link in body; got:\n%s", body)
+	if !strings.Contains(body, "/musicbrainz/?rule=rr-htmx") {
+		t.Errorf("expected BasePath-prefixed drill-down link to dashboard rule filter in body; got:\n%s", body)
```
[the remark this answers is not in the document]
Comment (sydlexius) [reply internal/api/handlers_report_test.go:229]: Fixed in e0e4aed. Added `TestHandleReportHealth_HTMX_DrillDownEscapesRuleID` seeding a rule ID with reserved characters; asserts both the encoded href and the absence of the raw form.
